samanthas_telegram_bot/api_queries/api_client.py

```diff
@@ -0,0 +1,371 @@
+        logger: Logger,
+        success_message: str,
+        failure_message: str,
+        success_logging_level: LoggingLevel = LoggingLevel.INFO,
+        failure_logging_level: LoggingLevel = LoggingLevel.ERROR,
+        notify_admins_mode: SendToAdminGroupMode = SendToAdminGroupMode.FAILURE_ONLY,
+        parse_mode_for_admin_group_message: ParseMode | None = None,
+    ) -> DataDict | None:
+        """Sends a request to backend, returns data, logs and notifies admins of potential errors.
+
+        Returns data received from backend in response.
+
+        This method is supposed to be called from within a chat (e.g. from a callback that has
+        a context object passed to it).
+        """
+        response = await cls._make_async_request(method=method, url=url, data=data)
+
+        message_prefix = f"Chat {context.user_data.chat_id}: "  # type: ignore[attr-defined]
+        message_suffix = ""
+        failure_message_suffix = (
+            f"{message_suffix} {response.status_code=} {response.json()=} {response.content=} "
+            f"@{os.environ.get('BOT_OWNER_USERNAME')}"
+        )
+
+        if parse_mode_for_admin_group_message == ParseMode.MARKDOWN_V2:
+            message_prefix = escape_for_markdown(message_prefix)
+            message_suffix = escape_for_markdown(message_suffix)
+            failure_message_suffix = escape_for_markdown(failure_message_suffix)
+
+        if response.status_code != expected_status_code:
+            await log_and_notify(
+                bot=context.bot,  # type: ignore[attr-defined]
+                logger=logger,
+                level=failure_logging_level,
+                text=f"{message_prefix}{failure_message}{failure_message_suffix}",
+                needs_to_notify_admin_group=notify_admins_mode
+                in (SendToAdminGroupMode.FAILURE_ONLY, SendToAdminGroupMode.SUCCESS_AND_FAILURE),
+                parse_mode_for_admin_group_message=parse_mode_for_admin_group_message,
+            )
+            return None
+
+        await log_and_notify(
+            bot=context.bot,  # type: ignore[attr-defined]
+            logger=logger,
+            level=success_logging_level,
+            text=f"{message_prefix}{success_message}{message_suffix}",
+            needs_to_notify_admin_group=notify_admins_mode
+            in (SendToAdminGroupMode.SUCCESS_ONLY, SendToAdminGroupMode.SUCCESS_AND_FAILURE),
+            parse_mode_for_admin_group_message=parse_mode_for_admin_group_message,
+        )
+
+        try:
+            response_json = response.json()
+        except AttributeError as e:
+            exc_info = f" {str(e)}"
+            if parse_mode_for_admin_group_message == ParseMode.MARKDOWN_V2:
+                failure_message_suffix += escape_for_markdown(exc_info)
+            else:
+                failure_message_suffix += exc_info
+            return None
+        else:
+            logger.debug(f"Response JSON: {response_json}")
```

Comment:
Oh. You already have log of response here. So I'm not sure if you need any additional logs.

Reply:
This is a `logger.debug()` :) it won't be shown in normal mode